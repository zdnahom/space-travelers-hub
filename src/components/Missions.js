import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getMissions } from "../redux/features/missions/missionsSlice";

const Missions = () => {
  const { missions, isLoading } = useSelector((store) => store.missions);
  console.log(missions);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getMissions());
  }, []);
  return (
    <section>
      {isLoading ? (
        <div className="loading">Loading...</div>
      ) : (
        <div>
          <table>
            <tr>
              <th>Mission</th>
              <th>Description</th>
              <th>Status</th>
              <th></th>
            </tr>
            {missions.map((mission) => (
              <tr>
                <td>{mission.mission_name}</td>
                <td>{mission.description}</td>
                <td>Not a member</td>
                <td> <button type="button">Join Mission</button></td>
              </tr>
            ))}
            </table>
        </div>
      )}
    </section>
  );
};
export default Missions;
