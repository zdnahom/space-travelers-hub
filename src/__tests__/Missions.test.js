import { render, cleanup, screen } from "@testing-library/react";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import configureMockStore from "redux-mock-store";
import Missions from "../components/Missions";

const mockStore = configureMockStore();
const store = mockStore({
  missions: {
    missions: [
      {
        mission_id: "1",
        mission_name: "Go to Mars",
        description: "Mars is a great planet",
        reserved: false,
      },
      {
        mission_id: "2",
        mission_name: "Go to Jupiter",
        description: "Jupiter is a great planet",
        reserved: false,
      },
      {
        mission_id: "3",
        mission_name: "Go to Saturn",
        description: "Saturn is a great planet",
        reserved: false,
      },
    ],
    isFetched: true,
  },
});
afterEach(cleanup);

describe("Mission component test", () => {
  it("It renders the missions", () => {
    render(
      <Provider store={store}>
        <Missions />
      </Provider>
    );
    const mission = screen.getByText("Go to Mars");
    expect(mission).toBeInTheDocument()
  });
  it("render mission lists", () => {
    const {container} = render(
      <Provider store={store}>
        <Missions />
      </Provider>
    );
    expect(container).toMatchSnapshot()
  });
});
