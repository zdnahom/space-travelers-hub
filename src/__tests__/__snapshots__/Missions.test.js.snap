// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission component test Join mission  1`] = `
<div>
  <section
    class="missions-container"
  >
    <div>
      <table
        class="missions"
      >
        <thead>
          <tr
            class="mission-table-row"
          >
            <th
              class="mission-table-header"
            >
              Mission
            </th>
            <th
              class="mission-table-header"
            >
              Description
            </th>
            <th
              class="mission-table-header"
            >
              Status
            </th>
            <th
              class="mission-table-header"
            >
               
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Mars
            </td>
            <td
              class="mission-table-data"
            >
              Mars is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Jupiter
            </td>
            <td
              class="mission-table-data"
            >
              Jupiter is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Saturn
            </td>
            <td
              class="mission-table-data"
            >
              Saturn is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;

exports[`Mission component test Join mission 1`] = `
<div>
  <section
    class="missions-container"
  >
    <div>
      <table
        class="missions"
      >
        <thead>
          <tr
            class="mission-table-row"
          >
            <th
              class="mission-table-header"
            >
              Mission
            </th>
            <th
              class="mission-table-header"
            >
              Description
            </th>
            <th
              class="mission-table-header"
            >
              Status
            </th>
            <th
              class="mission-table-header"
            >
               
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Mars
            </td>
            <td
              class="mission-table-data"
            >
              Mars is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Jupiter
            </td>
            <td
              class="mission-table-data"
            >
              Jupiter is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Saturn
            </td>
            <td
              class="mission-table-data"
            >
              Saturn is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;

exports[`Mission component test render mission lists 1`] = `
<div>
  <section
    class="missions-container"
  >
    <div>
      <table
        class="missions"
      >
        <thead>
          <tr
            class="mission-table-row"
          >
            <th
              class="mission-table-header"
            >
              Mission
            </th>
            <th
              class="mission-table-header"
            >
              Description
            </th>
            <th
              class="mission-table-header"
            >
              Status
            </th>
            <th
              class="mission-table-header"
            >
               
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Mars
            </td>
            <td
              class="mission-table-data"
            >
              Mars is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Jupiter
            </td>
            <td
              class="mission-table-data"
            >
              Jupiter is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
        <tbody>
          <tr
            class="mission-table-row"
          >
            <td
              class="mission-table-data"
            >
              Go to Saturn
            </td>
            <td
              class="mission-table-data"
            >
              Saturn is a great planet
            </td>
            <td
              class="mission-table-data"
            >
              <span
                class="non-member"
              >
                NOT A MEMBER
              </span>
            </td>
            <td
              class="mission-table-data"
            >
              <button
                class="join-button"
                type="button"
              >
                Join Mission
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </section>
</div>
`;
